### PR TITLE
JoinFetch to Join translation for count queries with filtering applied to fetched associations

### PR DESCRIPTION
**Pull request by tkaczmarzyk**
…ount queries)

### CODE REVIEW

**Comment by tkaczmarzyk** [comment]
related to #138

**Comment by tkaczmarzyk** [comment]
Good first issue in terms of -- perhaps sb could implement more tests (e.g. for multi-level joins)

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/JoinFetchInCountQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+ * 
+ * @author Tomasz Kaczmarzyk
+ */
+public class JoinFetchInCountQueryTest extends IntegrationTestBase {
+
+    private static final String BLACK_FRIDAY_TAG = "BlackFridayOffer";
+    
+	Customer homerSimpson;
+    Customer margeSimpson;
+    Customer bartSimpson;
```
Comment (jradlica) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/JoinFetchInCountQueryTest.java:47]:
```suggestion
    Customer homerSimpson;
    Customer margeSimpson;
    Customer bartSimpson;
```

File: src/test/java/net/kaczmarzyk/spring/data/jpa/CustomerBuilder.java [hunk 2]
```diff
@@ -15,13 +15,14 @@
...
 
-import javax.persistence.EntityManager;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
 import java.util.Calendar;
 import java.util.UUID;
 
+import javax.persistence.EntityManager;
+
```
Comment (jradlica) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/CustomerBuilder.java:25]:
```suggestion
```

**Review by jradlica — APPROVED** [review]

**Pull request merged**